### PR TITLE
Made ReadProfilingKey method public and more generic

### PR DESCRIPTION
**Pull request by slinkydeveloper**
Related to work on https://github.com/knative/eventing/issues/2148

### CODE REVIEW

File: profiling/server.go [hunk 1]
```diff
@@ -91,7 +91,7 @@ func readProfilingFlag(configMap *corev1.ConfigMap) (bool, error) {
 // UpdateFromConfigMap modifies the Enabled flag in the Handler
 // according to the value in the given ConfigMap
 func (h *Handler) UpdateFromConfigMap(configMap *corev1.ConfigMap) {
-	enabled, err := readProfilingFlag(configMap)
+	enabled, err := ReadProfilingFlag(configMap.Data)
```
Comment (matzew) [inline profiling/server.go]: I think this is better!

**Comment by slinkydeveloper** [comment]
@vagababov does it looks good now?

**Comment by vagababov** [comment]
/approve
/lgtm
Might need someone else to approve

**Pull request merged**